test/klaviyo/write-delete/main.go

```diff
@@ -69,13 +56,9 @@ func main() {
 
 	newName := gofakeit.AppName()
 	updateTag(ctx, conn, tagID, &TagPayload{
-		Tag: Tag{
-			ID:   tagID,
-			Type: "tag",
-			Attributes: TagAttributes{
-				Name: newName,
-			},
-		},
+		ID:   tagID,
+		Type: "tag",
```

Comment:
We should not require the user to provide `Type` either. Since the object name is already captured in WriteParams

Also "ID" should be "Id", see https://docs.withampersand.com/define-integrations/write-actions#update-an-existing-record for the Write API request that customers make.

Reply:
I'm conditionally approving this, but please revise the code to be able to accept this as the TagPayload:

```
&TagPayload{
   Id: tagID,
   Name: newName,
 }
```

Reply:
Ok, it took some time to verify all exceptions. I added a new mapping between ObjectName and the Type property for Write payload.

The payload is even better, it is just 
```
&TagPayload{
   Name: newName,
 }
```
RecordId will be moved to the payload for all update cases automatically. This way payload holds essentials and whatever the logic in the server to populate WriteParams.RecordId will end up as "id" property in the payload an in the URL path.